mlx/backend/common/eigvalsh.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include "mlx/allocator.h"
+#include "mlx/backend/common/copy.h"
+#include "mlx/linalg.h"
+#include "mlx/primitives.h"
+
+#ifdef ACCELERATE_NEW_LAPACK
+#include <Accelerate/Accelerate.h>
+#else
+#include <lapack.h>
+#endif
+
+namespace mlx::core {
+
+namespace {
+
+// Delegate to the eigenvalue decomposition taking into account differences in
+// LAPACK implementations (basically how to pass the 'jobz' and 'uplo' strings
+// to fortran).
+int ssyevd_wrapper(char jobz, char uplo, float* matrix, float* w, int N) {
```

Comment:
I don't think it needs to be in this PR, but it would be nice to add support for arbitrary matrices (not just hermitian/symmetric ones).

Hopefully that could just be a flag on the `Eig` primitive that then uses a different `lapack` incantation?